Lesson18/Ayse/script.js

```diff
@@ -0,0 +1,403 @@
+/*
+===========================================================
+  SHOPPING CART APPLICATION
+===========================================================
+In this project, you'll create a simple Shopping Cart to
+simulate adding items, removing items, calculating totals,
+and applying discounts.
+
+You'll practice:
+1. Classes and objects
+2. Encapsulation and abstraction
+3. Methods (functions inside a class)
+4. Arrays and basic array methods (push, filter, find)
+5. Conditional statements (if-else)
+
+Below is a step-by-step guide with comments explaining
+each part. You can test each step by running the code in
+Node.js or a browser console.
+*/
+
+/*
+-----------------------------------------------------------
+  STEP 1: Create the ShoppingCart Class
+-----------------------------------------------------------
+1. Define a `ShoppingCart` class.
+2. Add a constructor that initializes an empty private
+   array `_items` to store the cart items.
+3. Add a `viewCart` method to display all items in the cart.
+*/
+
+// 1. Create the ShoppingCart Class
+console.log('---------- 1: Create the ShoppingCart Class ----------');
+
+class ShoppingCart {
+
+  constructor() {
+    this._items = [];
+  }
+
+  viewCart() {
+    if (this._items.length === 0) {
+      console.log('Cart is empty');
+    }else {
+      console.log('Your cart contains: ', this._items);
+    }
+  }
+}
+const myCart = new ShoppingCart();
+myCart.viewCart();
+
+
+/*
+-----------------------------------------------------------
+  STEP 2: Add Items to the Cart
+-----------------------------------------------------------
+1. Create an `addItem` method in the `ShoppingCart` class.
+2. The method should:
+   - Accept `name`, `price`, and `quantity` as parameters.
+   - Check if the item already exists in the cart.
+     - If it exists, increase the quantity.
+     - Otherwise, add the new item to the `_items` array.
+*/
+// 2. Add Items to the Cart
+console.log('---------- 2: Add Items to the Cart ----------');
+
+class ShoppingCart2 {
```

Comment:
You don't need to define another ShoppingCart class, use the same class for the entire assignment.